models/users.js

```diff
@@ -0,0 +1,72 @@
+var mongoose = require('mongoose');
+var bcrypt = require('bcryptjs');
+
+//user schema
+var UserSchema = mongoose.Schema({
+    username: {
+        type: String,
+        unique:true,
+        index: true
+    },
+    name: {
+        type: String
+    },
+    email: {
+        type: String
+    },
+    phone: {
+        type: String
+    },
+    password: {
+        type: String
+    },
+    password2: {
+        type: String
+}
+
+})
+
+//create a model of the schema
+var User = module.exports = mongoose.model('User', UserSchema);
+
+//hash password 
+module.exports.createUser = function (newUser, callback) {
+    bcrypt.genSalt(10, function (err, salt) {
+        bcrypt.hash(newUser.password, salt, function (err, hash) {
+            // Store hash in your password DB. 
+            newUser.password = hash;
+            newUser.save(callback);
+        });
+    });
+}
+//custom functions for accessing the database
```

Comment:
consider creating a `helpers` directory and all these functions can live there.

Reply:
Updated to reflect a new helpers dir